src/core/components/layout/components/columns/styles.ts

```diff
@@ -135,7 +137,62 @@ const generateWidthCSS = (width: number | number[]) => {
 	return calculateWidth(width);
 };
 
-export const column = (width?: number | number[]) => {
+const generateSpanCSS = (span: number | number[]) => {
+	if (Array.isArray(span)) {
+		const breakpoints: Breakpoint[] = [
+			'mobile',
+			'tablet',
+			'desktop',
+			'leftCol',
+			'wide',
+		];
+
+		return span.reduce((styles, w, i) => {
+			return css`
+				${styles}
+				${from[breakpoints[i]]} {
+					${calculateSpan(w)};
+				}
+			`;
+		}, css``);
+	}
+
+	return calculateSpan(span);
+};
+
+const columnBreakpoints: Array<ColumnBreakpoint> = [
+	{ totalColumns: 4, rule: until.tablet },
+	{ totalColumns: 12, rule: between.tablet.and.leftCol },
+	{ totalColumns: 14, rule: between.leftCol.and.wide },
+	{ totalColumns: 16, rule: from.wide },
+];
+
+const calculateSpan = (span: number) => {
+	const columnBreakpointCss = columnBreakpoints.reduce(
+		(acc: string, cur: ColumnBreakpoint) => {
+			if (span === 0) return `display: none;`;
+			if (span >= cur.totalColumns) return `width: 100%; display: block;`;
+
+			const inferredWidth = span / cur.totalColumns;
+			const cssForBreakpoint = `{ width: calc((100% + ${space[5]}px) * ${inferredWidth} - ${space[5]}px); display: block; }\n`;
```

Comment:
Not sure, but is it possible to use the existing [`calculateWidth`](https://github.com/guardian/source/blob/237dcb06cb549af69036f194560dc224288ef2ec/src/core/components/layout/components/columns/styles.ts#L93) function to do this?

Reply:
Great idea, I seem to have made it work.